src/nvim/os/input.c

```diff
@@ -320,20 +320,21 @@ static void process_interrupts(void)
     return;
   }
 
-  char *inbuf = rbuffer_read_ptr(input_buffer);
-  size_t count = rbuffer_pending(input_buffer), consume_count = 0;
+  RBUFFER_UNTIL_READ_OR_EMPTY(input_buffer, inbuf, count) {
```

Comment:
That does not work if the buffer wraps around and in the first half (of the read section) a `3` was found:
In the first loop iteration the first half is cleared up to the `3`. The next iteration searches in the remainder of the first half. But because we do at most two iterations the second half is not searched for a `3`.